source/onboard/migrate-from-slack.rst

```diff
@@ -0,0 +1,191 @@
+Migrate from Slack
+==================
+
+.. include:: ../_static/badges/allplans-selfhosted.rst
+  :start-after: :nosearch:
+
+To migrate from Slack to Mattermost involves the following steps:
+
+1. `Prepare your Mattermost server <#prepare-your-mattermost-server>`__
+2. `Generate a Slack import <#generate-a-slack-import>`__
+3. `Download file attachments and email addresses <#download-file-attachments-and-email-addresses>`__
+4. `Convert the Slack import to Mattermost's format <#convert-slack-import-to-mattermost-s-bulk-export-format>`__
+5. `Import data into Mattermost <#import-data-into-mattermost>`__
+
+.. note::
+
+  If you are migrating to a self-hosted deployment of Mattermost, we recommend configuring your Mattermost server to use an S3 filestore before migration, as opposed to using the local filesystem filestore. If you are testing this in a development environment, you can use the ``minio`` docker image to simulate using an S3 bucket.
+
+1. Prepare your Mattermost server
+---------------------------------
+
+We recommend you create a separate team in Mattermost for each of your Slack workspaces.
+
+You can import two workspaces into the same team, but you'll need to ensure there are no channel name collisions first. Make sure that all users in Mattermost have the same username as in Slack, otherwise the import will fail.
+
+Also, system administrator roles will be overwritten if the usernames match and the user isn't an admin on the Slack workspace.
+
+2. Generate a Slack import
+--------------------------
+
+Slack offers two ways to `export your data from their product <https://get.slack.help/hc/en-us/articles/201658943-Export-your-workspace-data>`_.
+
+1. Regular export - Contains only public channel posts. This does not include private channels, DMs, or group conversations. This can be generated from **Slack > Administration > Workspace settings > Import/Export Data > Export > Start Export**.
```

Comment:
Getting to the import/export button wasn't that easy. The correct path was:

```
Workspace menu > Tools & Settings > Workspace Settings > Import/Export data button > Export 
```

It wasn't trivial even coming back to that section, as I went to _Customize workspace_ rather than _Workspace settings_ after actually knowing where the button was.

Maybe we can put some screenshots here? I made some.

cc: @cwarnermm

Reply:
Screenshots are always a good addition to docs. They help orient users in our product and validate that the user is in the correct place for the next step in a process. All for adding screenshots and I can help with this lift, once we've landed the PR.

Reply:
I'm trying to get some screenshots but unfortunately the Slack webapp keeps crashing

> Maybe we can put some screenshots here? I made some.

@fmartingr Do you mean that you have screenshots to share?

Reply:
> I'm trying to get some screenshots but unfortunately the Slack webapp keeps crashing
> 
> > Maybe we can put some screenshots here? I made some.
> 
> @fmartingr Do you mean that you have screenshots to share?

Yes, I will share them with you in Community.